### ISSUE TITLE
Apple Silicon Support for M1/ARM?

### ISSUE THREAD

**Issue by nickv2002** [ImageOptim/ImageOptim#334]
Thanks for the great project!

Is there a Universal version or Apple Silicon version available that will support M1 Chips on the ARM Mac's natively?

(I'll bet it would run a good deal faster 😁)

**Comment by kornelski** [comment]
Yes, there is: https://imageoptim.com/changelog.html

**Comment by andrewcrook** [comment]
@nickv2002  It would also depend on all the other utilities used to also be updated to get the full benefits

**Comment by kornelski** [comment]
These tools don't have to do anything. Support isn't added in the tools, but in compilers/build systems.

For oxipng it needs: https://github.com/rust-lang/cargo/issues/8875

and other tools already have support in Apple-llvm, so the support is done.

**Comment by kornelski** [comment]
However, I haven't finished M1 release of ImageOptim, because my M1 MacBook Pro stopped working completely. macOS Big Sur is so buggy that it bricked the entire machine to the point it won't even reinstall (hey, did you know there's no nvram reset on M1? And it has iOS-like on-line verification that can block system from being reinstalled? I know now). 

So for now I only have an Intel Mac working, and big distaste for M1/Big Sur from wasting a week fighting with its iOS-like locked-down buggy mess.

**Comment by andrewhag** [comment]
Sorry to hear about your M1 problems @kornelski! That's a bummer. I tried the beta but on open it showed a dialog that said "… can't be opened because Apple cannot check it for malicious software. (This software needs to be updated.)" In the meantime the Intel version is working fine on my M1. Thanks for your hard work on this app!

**Comment by kornelski** [comment]
Oof, that's a notarization issue.

**Comment by andrewcrook** [comment]
> macOS Big Sur is so buggy that it bricked the entire machine to the point it won't even reinstall (hey, did you know there's no nvram reset on M1? And it has iOS-like on-line verification that can block system from being reinstalled? I know now).

I heard there was a M1 big sur reinstall bug that has since been fixed. SMC and all that does have command line tools to configure. But don't try to reset them as you can only restore default profiles from another Mac + USB-C cable + configurator 2

**Comment by aasun** [comment]
Checking on this?

**Comment by reden87** [comment]
Is there any news on the release @kornelski ? 🙏

**Comment by kg-currenxie** [comment]
1.8.8 works for me. M1 + Monterey.

**Comment by dannyvargas23** [comment]
> 1.8.8 works for me. M1 + Monterey.

Yes, it works but it runs in emulation. The beta version is running natively.

**Comment by kg-currenxie** [comment]
Oh I see :) Thanks for clarifying-

**Comment by sdavids** [comment]
Any update?

**Comment by kornelski** [comment]
Xcode and Notarization toolign are shitty, and I don't think I will ever be able to make a working ImageOptim release again.

**Comment by kornelski** [comment]
I've verified on a clean/minimal project that Xcode is unable to correctly sign and Notarize any macOS projects that contain bundled command-line executables (in Xcode-speak: macOS App project can't depend on a Command-Line Tool target). 

ImageOptim is fundamentally a wrapper around many command-line executables, so this bug in Xcode is catastrophic for the project. 

**Until Apple fixes Xcode bug FB9885692, ImageOptim app is dead**. I'm prevented by Apple from making any new releases of the project. If I don't Notarize new releases, macOS will block them as malware. But I can't Notarize any new releases, because Xcode is either too buggy, or worse, intentionally dropped support for signing and notarizing of applications that use command-line executables.

**Comment by alecazam** [comment]
This was good to know, I have a tool called kram, that I also need to embed in an application but I haven't attempted it yet.  I just came across this the other day, but you probably have already read through this.  

https://developer.apple.com/documentation/xcode/embedding-a-helper-tool-in-a-sandboxed-app

**Comment by nickv2002** [comment]
I'm not a Mac developer but I do know there are many Mac apps that wrap CLI tools – `git` and `ffmpeg` are popular things to wrap with a (notarized) GUI app.

[GitHub Desktop](https://github.com/desktop/desktop) does this if anyone wants to look at their source, but maybe they build a signed version of `git` and that works?

That said I respect kornelski decision to walk away as long as the bug blocks his previous path.

**Comment by JokerQyou** [comment]
But according to [Apple documentation](https://developer.apple.com/documentation/xcode/embedding-a-helper-tool-in-a-sandboxed-app) it is feasible to wrap command-line tools in GUI apps:

> When building an app for the Mac, you can embed a command-line tool in the app to act as a helper tool.

**Comment by andrewcrook** [comment]
> But according to [Apple documentation](https://developer.apple.com/documentation/xcode/embedding-a-helper-tool-in-a-sandboxed-app) it is feasible to wrap command-line tools in GUI apps:
> 
> > When building an app for the Mac, you can embed a command-line tool in the app to act as a helper tool.

This is true because macOS is slowly getting rid of preinstalled languages like Ruby. The advice from Apple is if your application is using them to include specific versions with the application. JetBrains has done this with the Java JRE for a while with its products. The alternative is to tell the user to install them via homebrew or distribute the application via homebrew and have them as dependency homebrew packages. Another would be have the application download them into a support directory like iTerm2 does with Python. But I think Apple's approach is probably the better one.

**Comment by gingerbeardman** [comment]
> Why not release unnotarized? Doing one additional right-click is totally acceptable to use native instructions.

It is, as 1.9.0 beta

https://imageoptim.com/changelog.html

**Comment by minyuex** [comment]
@gingerbeardman @kornelski @sdavids @nickv2002 @andrewcrook Update to 1.9.5. Request to merge to the main branch. dmg download from below: https://github.com/minyuex/ImageOptim/releases/tag/v1.9.5
